sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -197,37 +205,100 @@ private void sendBody(HttpRequest httpRequest, HttpURLConnection connection) {
      * @return A HttpResponse object
      */
     private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection connection) {
-        try {
-            int responseCode = connection.getResponseCode();
-            HttpHeaders responseHeaders = getResponseHeaders(connection);
+        HttpHeaders responseHeaders = getResponseHeaders(connection);
+        HttpResponse<?> httpResponse = createHttpResponse(httpRequest, connection);
+
+        if (isTextEventStream(responseHeaders)) {
+            try {
```

Comment:
We do, because if we throw when listener is `null` we will leave the connection open. Wrapping that part in the try/finally should take care of it, right?